src/Aspire.Hosting/Dashboard/DashboardViewModelService.cs

```diff
@@ -90,6 +91,90 @@ await foreach (var tuple in _kubernetesService.WatchAsync<T>(cancellationToken:
         });
     }
 
+    /// <summary>
+    /// This is workaround for https://github.com/dotnet/aspire/issues/1364 where there is a timing issue where an existing connection to Traefik
+    /// will return a 404 for potentially several minutes if that connection is made before it is finished loading its configuration. The below code
+    /// will hit the proxy endpoint setup in that configuration until we do not see the known 404 response from Traefik.
+    /// </summary>
+    private void RunServiceWatchTask()
```

Comment:
Can you break this logic into multiple methods and deconstruct the tuples inline to remove Item1 and Item2 in the code?

Reply:
I've got a refactor almost ready to push